sloth/src/test/scala/SlothSpec.scala

```diff
@@ -56,7 +61,48 @@ object ApiImplFunResponse extends Api[ApiResultFun] {
   def multi(a: Int)(b: Int): ApiResultFun[Int] = i => ApiResult("hans", Future.successful(a + b + i))
 }
 
+// mixed result type
+trait MixedApi {
+  def fun: Future[Int]
+  def stream: Observable[Int]
+}
+object MixedApiImpl extends MixedApi {
+  def fun: Future[Int] = Future.successful(1)
+  def stream: Observable[Int] = Observable(1, 2)
+}
+
+
+// multi arg result type
+trait MultiTypeArgApi[Single[_], Stream[_]] {
+  def fun: Single[Int]
+  def stream: Stream[Int]
+}
+object MultiTypeArgApiImpl extends MultiTypeArgApi[Future, Observable] {
+  def fun = Future.successful(1)
+  def stream = Observable(1, 2)
+}
+
+// type fun
+trait TypeFunArgApi[Fun[F[_], _]] {
+  def fun: Task[Int]
+  def stream: Observable[Int]
+  def state: Fun[Task, String]
+  def sync: Int
+}
+object TypeFunArgApi {
+  type Client[F[_], T] = F[T]
+  type Server[F[_], T] = String => F[T]
+}
+object TypeFunArgApiImpl extends TypeFunArgApi[TypeFunArgApi.Server] {
+  def fun = Task.pure(0)
+  def stream = Observable(1, 2)
+  def state = state => Task.pure(state)
+  def sync = 5
+}
+
```

Comment:
These need better documentation.  
Currently this is only mentioned in `Readme.md#268`

Reply:
Agree, I can make another section in the readme that explains a more advanced encoding you could use (like this one).